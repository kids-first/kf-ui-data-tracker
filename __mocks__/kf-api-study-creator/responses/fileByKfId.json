{
  "data": {
    "fileByKfId": {
      "kfId": "SF_5ZPEM167",
      "name": "organization.jpeg",
      "description": "Month necessary animal end standard case. View system operation message decade. Actually sing because deal everything woman subject.",
      "fileType": "CLN",
      "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167",
      "versions": {
        "edges": [
          {
            "node": {
              "kfId": "FV_TT8NF09R",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_TT8NF09R",
              "size": 9893,
              "createdAt": "2018-08-21T03:16:11+00:00",
              "__typename": "ObjectNode"
            },
            "__typename": "ObjectNodeEdge"
          },
          {
            "node": {
              "kfId": "FV_DMTKEFQW",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_DMTKEFQW",
              "size": 5460,
              "createdAt": "2018-09-27T05:32:26+00:00",
              "__typename": "ObjectNode"
            },
            "__typename": "ObjectNodeEdge"
          },
          {
            "node": {
              "kfId": "FV_2WV3VNJN",
              "downloadUrl": "http://localhost:8080/download/study/SD_8WX8QQ06/file/SF_5ZPEM167/version/FV_2WV3VNJN",
              "size": 5419,
              "createdAt": "2017-08-25T00:05:30+00:00",
              "__typename": "ObjectNode"
            },
            "__typename": "ObjectNodeEdge"
          }
        ],
        "__typename": "ObjectNodeConnection"
      },
      "__typename": "FileNode"
    }
  }
}
