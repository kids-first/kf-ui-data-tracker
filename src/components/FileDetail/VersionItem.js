import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Mutation} from 'react-apollo';
import {FILE_DOWNLOAD_URL} from '../../state/mutations';
import {Avatar, Icon} from 'kf-uikit';
import TimeAgo from 'react-timeago';
import CopyButton from '../CopyButton/CopyButton';
import {
  fileTypeDetail,
  formatFileSize,
  downloadFile,
} from '../../common/fileUtils';
/**
 * Displays single version item from the list
 */
const VersionItem = ({
  className,
  studyId,
  fileId,
  fileType,
  fileName,
  versionNode,
  index,
}) => {
  const ext = fileName.substr(fileName.lastIndexOf('.'));
  let versionItemClass = classes('FileVersionList--Element', className);
  return (
    <Mutation mutation={FILE_DOWNLOAD_URL}>
      {downloadFileMutation => (
        <li key={versionNode.kfId} className={versionItemClass}>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 94 94"
            height="12"
            width="12"
            className="mx-16"
          >
            <defs />
            <title>icon-error</title>
            <g id="Layer_1" data-name="Layer 1">
              <circle fill="#fff" cx="47" cy="47" r="43" />
              <path
                fill="#e45562"
                d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
                transform="translate(0 0)"
              />
              <path
                fill="#e45562"
                d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
                transform="translate(0 0)"
              />
            </g>
          </svg>
          <button
            className="FileVersionElement--button w-24"
            onClick={e =>
              downloadFile(
                studyId,
                fileId,
                versionNode.kfId,
                downloadFileMutation,
              )
            }
          >
            {index > 0 ? versionNode.kfId : 'Latest'}
          </button>
          <CopyButton
            text={versionNode.kfId}
            className="FileVersionElement--button FileVersionList--hidden w-12"
          />
          <Avatar
            className="inline-block mr-8 hidden"
            size={20}
            imgUrl="https://www.w3schools.com/css/img_avatar.png"
          />
          <TimeAgo
            date={versionNode.createdAt}
            live={false}
            className="FileVersionList--text w-32"
          />
          <p className="FileVersionList--text w-16 hidden">{ext}</p>
          <p className="FileVersionList--text w-24">
            {formatFileSize(versionNode.size, true) || 'Size Unknown'}
          </p>
          <div className="flex">
            <div className="FileVersionList--icon">
              <Icon
                width={12}
                height={12}
                kind={fileTypeDetail[fileType].icon}
              />
            </div>
            <p className="FileVersionList--text my-0 w-48">
              {fileTypeDetail[fileType].title}
            </p>
          </div>
          <button
            onClick={e =>
              downloadFile(
                studyId,
                fileId,
                versionNode.kfId,
                downloadFileMutation,
              )
            }
          >
            <Icon
              kind="download"
              width={12}
              height={12}
              className="FileVersionList--hidden w-12"
            />
          </button>
        </li>
      )}
    </Mutation>
  );
};

VersionItem.propTypes = {
  /** Any additional classes to be applied to the study list*/
  className: PropTypes.string,
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File kfId*/
  fileId: PropTypes.string,
  /** File type*/
  fileType: PropTypes.string,
  /** File name*/
  fileName: PropTypes.string,
  /** Version object*/
  versionNode: PropTypes.object.isRequired,
  /** Index of version item (sort by createdAt)*/
  index: PropTypes.number,
};

VersionItem.defaultProps = {
  className: null,
  studyId: null,
  fileId: null,
  fileType: null,
  fileName: null,
  fileNode: {},
  index: null,
};

export default VersionItem;
