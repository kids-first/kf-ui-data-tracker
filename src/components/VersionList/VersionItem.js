import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {Mutation} from 'react-apollo';
import {FILE_DOWNLOAD_URL} from '../../state/mutations';
import {Avatar, Icon, GridContainer} from 'kf-uikit';
import TimeAgo from 'react-timeago';

import {
  fileTypeDetail,
  versionState,
  formatFileSize,
  downloadFile,
} from '../../common/fileUtils';
import SvgIcon from '../Icon/Icon';
/**
 * Displays single version item from the list
 */
const VersionItem = ({
  className,
  studyId,
  fileId,
  fileType,
  versionNode,
  index,
  onNameClick,
}) => {
  let versionItemClass = classes('FileVersionList--Element', className);
  const stateColor = versionNode.state
    ? versionState[versionNode.state].color
    : 'bg-lightGrey';
  const iconClass = classes(
    'FileVersionElement--Icon',
    'row-3',
    'cell-1',
    'lg:row-1',
    'lg:cell-1',
    stateColor,
  );
  return (
    <Mutation mutation={FILE_DOWNLOAD_URL}>
      {downloadFileMutation => (
        <li key={versionNode.kfId} className={versionItemClass}>
          <GridContainer collapsed="rows" className="p-4">
            <div className={iconClass}>
              <SvgIcon
                kind={fileTypeDetail[fileType].icon}
                width="14"
                height="14"
              />
            </div>
            {index === 0 && (
              <div className="FileVersionElement--Tag row-3 cell-2 lg:row-2 lg:cell-1">
                Latest
              </div>
            )}
            <div className="flex row-1 cell-12 lg:cell-7">
              <p
                className="FileVersionElement--button m-0 p-0 text-lg"
                onClick={e => onNameClick(versionNode, index)}
              >
                {versionNode.fileName}
              </p>
            </div>
            <div className="flex row-3 cell-9 lg:row-1 lg:cell-4 lg:justify-end align-center">
              <div className="flex align-center">
                <Avatar
                  className="inline-block mr-8"
                  size={20}
                  imgUrl={
                    versionNode.creator
                      ? versionNode.creator.picture
                      : 'https://www.w3schools.com/css/img_avatar.png'
                  }
                  userName={versionNode.creator && versionNode.creator.username}
                  userEmail={versionNode.creator && versionNode.creator.email}
                />
                <TimeAgo
                  date={versionNode.createdAt}
                  live={false}
                  className="FileVersionList--text  FileVersionList--Element-meta"
                />
              </div>
              <p className=" FileVersionList--text FileVersionList--Element-meta row-1 cell-2">
                {formatFileSize(versionNode.size, true) || 'Size Unknown'}
              </p>
              <button
                onClick={e =>
                  downloadFile(
                    studyId,
                    fileId,
                    versionNode.kfId,
                    downloadFileMutation,
                  )
                }
                className="FileVersionList--text row-1 cell-1"
              >
                <Icon kind="download" width={12} height={12} />
              </button>
            </div>

            <p className="FileVersionList--Element-desc FileVersionList--text row-2 cell-12 lg:cell-2-11 text-xs">
              {versionNode.description || 'No version summary available'}
            </p>
          </GridContainer>
        </li>
      )}
    </Mutation>
  );
};

VersionItem.propTypes = {
  /** Any additional classes to be applied to the study list*/
  className: PropTypes.string,
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File kfId*/
  fileId: PropTypes.string,
  /** File type*/
  fileType: PropTypes.string,
  /** Version object*/
  versionNode: PropTypes.object.isRequired,
  /** Index of version item (sort by createdAt)*/
  index: PropTypes.number,
};

VersionItem.defaultProps = {
  className: null,
  studyId: null,
  fileId: null,
  fileType: null,
  fileNode: {},
  index: null,
};

export default VersionItem;
