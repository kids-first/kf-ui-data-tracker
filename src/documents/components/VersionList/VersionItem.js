import React from 'react';
import PropTypes from 'prop-types';
import {useMutation} from '@apollo/client';
import {FILE_DOWNLOAD_URL} from '../../mutations';
import AvatarTimeAgo from '../../../components/AvatarTimeAgo/AvatarTimeAgo';
import {formatFileSize, downloadFile, lengthLimit} from '../../utilities';
import {Label, Icon, Table} from 'semantic-ui-react';
/**
 * Displays single version item from the list
 */
const VersionItem = ({
  studyId,
  fileId,
  fileType,
  versionNode,
  index,
  onNameClick,
}) => {
  const [downloadFileMutation] = useMutation(FILE_DOWNLOAD_URL);
  const size = versionNode.size
    ? formatFileSize(versionNode.size, true)
    : 'Size Unknown';
  return (
    <Table.Row
      onClick={e => onNameClick(versionNode, index)}
      className="version--item"
    >
      <Table.Cell>
        <p title={versionNode.fileName}>
          <b>{lengthLimit(versionNode.fileName, 70)}</b>
        </p>
        <p title={versionNode.description} data-testid="version-item">
          <small>
            {lengthLimit(versionNode.description, 110)}
            {versionNode.description.length === 0 &&
              'No version summary available'}
          </small>
        </p>
      </Table.Cell>
      <Table.Cell textAlign="right" verticalAlign="top" collapsing>
        <AvatarTimeAgo
          size="mini"
          creator={versionNode.creator}
          createdAt={versionNode.createdAt}
        />
        <Label
          basic
          size="mini"
          as="button"
          onClick={e => {
            e.stopPropagation();
            downloadFile(
              studyId,
              fileId,
              versionNode.kfId,
              downloadFileMutation,
            );
          }}
        >
          <Icon name="download" />
          {size}
        </Label>
      </Table.Cell>
    </Table.Row>
  );
};

VersionItem.propTypes = {
  /** Study kfId*/
  studyId: PropTypes.string,
  /** File kfId*/
  fileId: PropTypes.string,
  /** File type*/
  fileType: PropTypes.string,
  /** Version object*/
  versionNode: PropTypes.object.isRequired,
  /** Index of version item (sort by createdAt)*/
  index: PropTypes.number,
};

VersionItem.defaultProps = {
  studyId: null,
  fileId: null,
  fileType: null,
  fileNode: {},
  index: null,
};

export default VersionItem;
