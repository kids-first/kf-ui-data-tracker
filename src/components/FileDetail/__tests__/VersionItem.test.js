import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {MockedProvider} from 'react-apollo/test-utils';
import {render, cleanup} from 'react-testing-library';
import VersionItem from '../VersionItem';
import fileByKfId from './fileByKfId';

afterEach(cleanup);
it('renders correctly with 0 as index -- show latest', () => {
  const fileNode = fileByKfId[0].data.fileByKfId;
  const versionNode = fileNode.versions.edges[0].node;
  const tree = render(
    <MockedProvider>
      <MemoryRouter>
        <VersionItem
          key={0}
          studyId={'SD_00000000'}
          fileId={fileNode.kfId}
          fileType={fileNode.fileType}
          fileName={fileNode.name}
          versionNode={versionNode}
          index={0}
        />
      </MemoryRouter>
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();
  const button = tree.getByText(/Latest/);
  expect(button.innerHTML).toBe('Latest');
});

it('renders correctly with 1 as index -- show version kfId', () => {
  const fileNode = fileByKfId[0].data.fileByKfId;
  const versionNode = fileNode.versions.edges[0].node;
  const tree = render(
    <MockedProvider>
      <MemoryRouter>
        <VersionItem
          key={1}
          studyId={'SD_00000000'}
          fileId={fileNode.kfId}
          fileType={fileNode.fileType}
          fileName={fileNode.name}
          versionNode={versionNode}
          index={1}
        />
      </MemoryRouter>
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();
  const button = tree.getByText(/FV_BC7BWZ2W/);
  expect(button.innerHTML).toBe('FV_BC7BWZ2W');
});
