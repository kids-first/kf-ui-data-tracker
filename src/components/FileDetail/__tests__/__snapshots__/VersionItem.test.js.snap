// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with 0 as index -- show latest 1`] = `
<div>
  <li
    class="FileVersionList--Element"
  >
    <svg
      class="mx-16"
      height="12"
      viewBox="0 0 94 94"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <title>
        icon-error
      </title>
      <g
        data-name="Layer 1"
        id="Layer_1"
      >
        <circle
          cx="47"
          cy="47"
          fill="#fff"
          r="43"
        />
        <path
          d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
          fill="#e45562"
          transform="translate(0 0)"
        />
        <path
          d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
          fill="#e45562"
          transform="translate(0 0)"
        />
      </g>
    </svg>
    <button
      class="FileVersionElement--button w-24"
    >
      Latest
    </button>
    <span
      class="FileVersionElement--button FileVersionList--hidden w-12"
    >
      <button>
        <svg
          height="10"
          id="mock-icon"
          kind="copy"
          width="10"
        >
          <title>
            icon-
            copy
          </title>
        </svg>
      </button>
    </span>
    <div
      class="Avatar inline-block mr-8 hidden"
      style="height: 20px; width: 20px;"
    >
      <img
        alt="null avatar"
        class="Avatar-img"
        src="https://www.w3schools.com/css/img_avatar.png"
      />
    </div>
    <time
      class="FileVersionList--text w-32"
      datetime="2019-05-16T15:13:02.090Z"
      title="2019-05-16T15:13:02.090275+00:00"
    >
      3 weeks from now
    </time>
    <p
      class="FileVersionList--text w-16 hidden"
    >
      .jpg
    </p>
    <p
      class="FileVersionList--text w-24"
    >
      61.0 kB
    </p>
    <div
      class="flex"
    >
      <div
        class="FileVersionList--icon"
      >
        <svg
          height="12"
          id="mock-icon"
          kind="settings"
          width="12"
        >
          <title>
            icon-
            settings
          </title>
        </svg>
      </div>
      <p
        class="FileVersionList--text my-0 w-48"
      >
        Other
      </p>
    </div>
    <button>
      <svg
        class="FileVersionList--hidden w-12"
        height="12"
        id="mock-icon"
        kind="download"
        width="12"
      >
        <title>
          icon-
          download
        </title>
      </svg>
    </button>
  </li>
</div>
`;

exports[`renders correctly with 1 as index -- show version kfId 1`] = `
<div>
  <li
    class="FileVersionList--Element"
  >
    <svg
      class="mx-16"
      height="12"
      viewBox="0 0 94 94"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <title>
        icon-error
      </title>
      <g
        data-name="Layer 1"
        id="Layer_1"
      >
        <circle
          cx="47"
          cy="47"
          fill="#fff"
          r="43"
        />
        <path
          d="M47,94A47,47,0,1,1,94,47,47,47,0,0,1,47,94ZM47,8A39,39,0,1,0,86,47,39,39,0,0,0,47,8Z"
          fill="#e45562"
          transform="translate(0 0)"
        />
        <path
          d="M59.83,64.07a4.25,4.25,0,0,1-3-1.26L47,53.05,37.2,62.83a4.32,4.32,0,0,1-6.05,0,4.26,4.26,0,0,1,0-6L41,47l-9.79-9.8a4.3,4.3,0,0,1,0-6.05,4.38,4.38,0,0,1,6.07,0L47.06,41l9.7-9.65a4.3,4.3,0,0,1,6,0,4.24,4.24,0,0,1,0,6.05L53.11,47l9.74,9.74a4.3,4.3,0,0,1,0,6.06,4.25,4.25,0,0,1-3,1.26Zm-1-30.75h0Z"
          fill="#e45562"
          transform="translate(0 0)"
        />
      </g>
    </svg>
    <button
      class="FileVersionElement--button w-24"
    >
      FV_BC7BWZ2W
    </button>
    <span
      class="FileVersionElement--button FileVersionList--hidden w-12"
    >
      <button>
        <svg
          height="10"
          id="mock-icon"
          kind="copy"
          width="10"
        >
          <title>
            icon-
            copy
          </title>
        </svg>
      </button>
    </span>
    <div
      class="Avatar inline-block mr-8 hidden"
      style="height: 20px; width: 20px;"
    >
      <img
        alt="null avatar"
        class="Avatar-img"
        src="https://www.w3schools.com/css/img_avatar.png"
      />
    </div>
    <time
      class="FileVersionList--text w-32"
      datetime="2019-05-16T15:13:02.090Z"
      title="2019-05-16T15:13:02.090275+00:00"
    >
      3 weeks from now
    </time>
    <p
      class="FileVersionList--text w-16 hidden"
    >
      .jpg
    </p>
    <p
      class="FileVersionList--text w-24"
    >
      61.0 kB
    </p>
    <div
      class="flex"
    >
      <div
        class="FileVersionList--icon"
      >
        <svg
          height="12"
          id="mock-icon"
          kind="settings"
          width="12"
        >
          <title>
            icon-
            settings
          </title>
        </svg>
      </div>
      <p
        class="FileVersionList--text my-0 w-48"
      >
        Other
      </p>
    </div>
    <button>
      <svg
        class="FileVersionList--hidden w-12"
        height="12"
        id="mock-icon"
        kind="download"
        width="12"
      >
        <title>
          icon-
          download
        </title>
      </svg>
    </button>
  </li>
</div>
`;
